src/Sylius/Component/Core/Taxation/OrderItemsTaxesByZoneApplicator.php

```diff
@@ -80,15 +80,10 @@ public function apply(OrderInterface $order, ZoneInterface $zone)
             $totalTaxAmount = $this->calculator->calculate($item->getTotal(), $taxRate);
             $splitTaxes = $this->distributor->distribute($totalTaxAmount, $quantity);
 
-            $units = $item->getUnits();
-            $units->first();
-            foreach ($splitTaxes as $key => $tax) {
-                if (0 === $tax) {
-                    continue;
-                }
-
-                $unit = $this->getNextUnit($units);
-                $this->addAdjustment($unit, $tax, $taxRate->getLabel(), $taxRate->isIncludedInPrice());
+            $i = 0;
+            foreach ($item->getUnits() as $unit) {
+                $this->addAdjustment($unit, $splitTaxes[$i], $taxRate->getLabel(), $taxRate->isIncludedInPrice());
```

Comment:
does this handle the previous check for `if (0 === $tax)` ? not sure what that was doing.